packages/material-icon/index.js

```diff
@@ -2,34 +2,80 @@ import React from 'react';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
 
+import withRipple from '@material/react-ripple';
+
 export default class MaterialIcon extends React.Component {
   render() {
     const {
       icon,
-      className,
+      hasRipple,
       ...otherProps
     } = this.props;
-    const classes = classnames('material-icons', {
-      [className]: className.length > 0,
-    });
+
+    if (hasRipple) {
+      return (
+        <RippleMaterialIcon
+          unbounded
+          icon={icon}
+          {...otherProps}
+        />
+      );
+    }
 
     return (
-      <i
-        className={classes}
+      <MaterialIconDefault
+        icon={icon}
         {...otherProps}
-      >
-        {icon}
-      </i>
+      />
     );
   }
 }
 
+const MaterialIconDefault = (props) => {
+  const {
+    className,
+    icon,
+    initRipple,
+    hasRipple, // eslint-disable-line no-unused-vars
+    ...otherProps
+  } = props;
+  const classes = classnames('material-icons', {
+    [className]: className.length > 0,
+  });
+
+  return (
+    <i
+      className={classes}
+      ref={initRipple}
+      {...otherProps}
+    >
+      {icon}
+    </i>
+  );
+};
+
+export const RippleMaterialIcon = withRipple(MaterialIconDefault);
+
 MaterialIcon.propTypes = {
   icon: PropTypes.string,
-  className: PropTypes.string,
+  hasRipple: PropTypes.bool,
 };
 
 MaterialIcon.defaultProps = {
+  icon: '',
+  hasRipple: false,
+};
+
+MaterialIconDefault.propTypes = {
```

Comment:
You don't actually export MaterialIconDefault, so I dont know if you need to specify propTypes.

Reply:
Lint yells at me otherwise :|